dlt-connector/src/client/BackendClient.ts

```diff
@@ -0,0 +1,88 @@
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+import { gql, GraphQLClient } from 'graphql-request'
+
+import { CONFIG } from '@/config'
+import { CommunityDraft } from '@/graphql/input/CommunityDraft'
+import { logger } from '@/logging/logger'
+import { LogError } from '@/server/LogError'
+
+const communityByForeign = gql`
+  query ($foreign: Boolean) {
+    community(foreign: $foreign) {
+      uuid
+      foreign
+      creationDate
+    }
+  }
+`
+interface Community {
+  community: {
+    uuid: string
+    foreign: boolean
+    creationDate: string
+  }
+}
+// Source: https://refactoring.guru/design-patterns/singleton/typescript/example
+// and ../federation/client/FederationClientFactory.ts
+/**
+ * A Singleton class defines the `getInstance` method that lets clients access
+ * the unique singleton instance.
+ */
+// eslint-disable-next-line @typescript-eslint/no-extraneous-class
+export class BackendClient {
+  // eslint-disable-next-line no-use-before-define
+  private static instance: BackendClient
+  client: GraphQLClient
+  /**
+   * The Singleton's constructor should always be private to prevent direct
+   * construction calls with the `new` operator.
+   */
+  // eslint-disable-next-line no-useless-constructor, @typescript-eslint/no-empty-function
+  private constructor() {}
+
+  /**
+   * The static method that controls the access to the singleton instance.
+   *
+   * This implementation let you subclass the Singleton class while keeping
+   * just one instance of each subclass around.
+   */
+  public static getInstance(): BackendClient | undefined {
+    if (!BackendClient.instance) {
+      BackendClient.instance = new BackendClient()
+    }
+    if (!BackendClient.instance.client) {
+      try {
+        BackendClient.instance.client = new GraphQLClient(CONFIG.BACKEND_SERVER_URL, {
+          headers: {
+            'content-type': 'application/json',
+          },
+          method: 'POST',
+          jsonSerializer: {
+            parse: JSON.parse,
+            stringify: JSON.stringify,
+          },
+        })
+      } catch (e) {
+        logger.error("couldn't connect to backend: ", e)
+        return
+      }
+    }
+    return BackendClient.instance
+  }
+
+  public async homeCommunityUUid(): Promise<CommunityDraft> {
+    logger.info('check home community on backend')
+    const { data, errors } = await this.client.rawRequest<Community>(communityByForeign, {
+      foreign: false,
+    })
+    if (errors) {
+      throw new LogError('error getting home community from backend', errors)
+    }
+    const communityDraft = new CommunityDraft()
+    communityDraft.uuid = data.community.uuid
+    communityDraft.foreign = data.community.foreign
+    communityDraft.createdAt = data.community.creationDate
+    return communityDraft
+  }
```

Comment:
As you see here in your dlt-code it is helpfull to name the methods more expressive like this example shows `homeCommunityUuid` than to generalize all to one `community` method with a very complex parameter treatment for all variants.

So on my opinion this method expresses the opposite of your refactorings in the `util.communities.ts`-file.

Reply:
The function names should show what the function is doing.
But a disadvantage is that is easy to forget to update the name, if the function changed. 
homeCommunityUUid isn't quite correct any longer here, because it returns not only the community uuid. I have updated it now. 

my refactor in community had also the goal to make the mysql request shorter. If I had not put that in one generalized function, It would increased the Lines of Codes quite a bit.
Without putting it into getCommunityFindOptions I would have add the if else structure to reduce mysql code in isHomeCommunity, getCommunityUrl, isCommunityAuthenticated and getCommunityName.

Reply:
The community grapqhl request had this name because graphql is designed for call for different datas via the same queries. I'd like to use graphQL as intended to make maximal use of the additional ressource cost using it instead of REST or jsonRPC. But even in popular REST Apis like from youtube they are normally not many different requests and mostly with easy and short names. 
- https://developers.google.com/youtube/v3/docs?hl=de
- https://www.apollographql.com/blog/8-free-to-use-graphql-apis-for-your-projects-and-demos
- https://docs.github.com/de/graphql/overview/public-schema